WordPress/Classes/Utility/Spotlight/SearchManager.swift

```diff
@@ -252,37 +252,45 @@ fileprivate extension SearchManager {
                    blogID: NSNumber,
                    onSuccess: @escaping (_ post: AbstractPost) -> Void,
                    onFailure: @escaping () -> Void) {
-        let context = ContextManager.sharedInstance().mainContext
+        let coreDataStack = ContextManager.shared
 
-        guard let blog = Blog.lookup(withID: blogID, in: context) else {
+        guard let blog = Blog.lookup(withID: blogID, in: coreDataStack.mainContext) else {
                 onFailure()
                 return
         }
 
-        let postService = PostService(managedObjectContext: context)
-        postService.getPostWithID(postID, for: blog, success: { apost in
-            onSuccess(apost)
-        }, failure: { error in
-            onFailure()
-        })
+        let postRepository = PostRepository(coreDataStack: coreDataStack)
+        Task { @MainActor in
+            do {
+                let postObjectID = try await postRepository.getPost(withID: postID, from: .init(saved: blog))
+                let post = try coreDataStack.mainContext.existingObject(with: postObjectID)
```

Comment:
Would it make sense to store `coreDataStack.mainContext` in a `let` since it's used both here and above to lookup the blog object? Just to ensure the used context stays the same between calls. 

Or, is it overengineering?

Reply:
Probably not overengineering, but it isn't necessary I guess. They don't have to use the same context though. The important thing is the mainContext is used in this Task.